cloudevents-sdk-rdkafka/src/kafka_consumer_record.rs

```diff
@@ -0,0 +1,264 @@
+use bytes::Bytes;
+use cloudevents::event::SpecVersion;
+use cloudevents::message::{
+    BinaryDeserializer, BinarySerializer, Encoding, MessageAttributeValue, MessageDeserializer,
+    Result, StructuredDeserializer, StructuredSerializer,
+};
+use cloudevents::{message, Event};
+use rdkafka::message::{BorrowedMessage, Headers, Message, OwnedMessage};
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::str;
+
+pub struct ConsumerRecordDeserializer {
+    pub(crate) headers: HashMap<String, Bytes>,
```

Comment:
For the header values and the payload, more than copying the bytes into `Bytes`, can we store the `&[u8]`? With some lifetimes, that should be doable because `ConsumerRecordDeserializer` for sure lives less than both `OwnedMessage` and `&BorrowedMessage`

Reply:
Ok now that i look at it, it doesn't really make much difference because then you buffer everything in order to go to event... I wonder, can you buffer to `Vec<u8>` more than using the `Bytes` struct? `Bytes` is meant to be a flexible byte buffer, we don't need its functionalities here

Reply:
Yeah on it. I put bytes cause i didn't really know at the time if i would need to do said fancy stuff, and then i completely forgot about it.